bson/src/main/scala/bson.scala

```diff
@@ -17,6 +17,7 @@ package reactivemongo.bson
 
 object `package` extends DefaultBSONHandlers {
   type BSONElement = (String, BSONValue)
+  type BSONCodec[A] = BSONDocumentReader[A] with BSONDocumentWriter[A] with BSONHandler[BSONDocument, A]
```

Comment:
I think `BSONDocumentHandler` is more consistent with `BSONHandler`.  @aloiscochard What do you think?